container/push.bzl

```
@@ -132,105 +230,233 @@ def _impl(ctx):
+    "repository_file": attr.label(
+        allow_single_file = True,
+        doc = "The label of the file with repository value. Overrides 'repository'.",
+    ),
+    "skip_unchanged_digest": attr.bool(
+        default = False,
+        doc = "Check if the container registry already contain the image's digest. If yes, skip the push for that image. " +
+              "Default to False. " +
+              "Note that there is no transactional guarantee between checking for digest existence and pushing the digest. " +
+              "This means that you should try to avoid running the same container_push targets in parallel.",
+    ),
+    "stamp": STAMP_ATTR,
+    "tag": attr.string(
+        default = "latest",
+        doc = "The tag of the image.",
+    ),
+    "tag_file": attr.label(
+        allow_single_file = True,
+        doc = "The label of the file with tag value. Overrides 'tag'.",
+    ),
+    "tag_tpl": attr.label(
+        mandatory = True,
+        allow_single_file = True,
+        doc = "The script template to use.",
+    ),
+    "windows_paths": attr.bool(
+        mandatory = True,
+    ),
+}, _layer_tools)
+
+container_push_ = rule(
+    attrs = dicts.add(_container_push_common_attrs, {
         "image": attr.label(
             allow_single_file = [".tar"],
             mandatory = True,
             doc = "The label of the image to push.",
         ),
-        "registry": attr.string(
-            mandatory = True,
-            doc = "The registry to which we are pushing.",
-        ),
-        "repository": attr.string(
-            mandatory = True,
-            doc = "The name of the image.",
-        ),
-        "repository_file": attr.label(
-            allow_single_file = True,
-            doc = "The label of the file with repository value. Overrides 'repository'.",
-        ),
-        "skip_unchanged_digest": attr.bool(
-            default = False,
-            doc = "Check if the container registry already contain the image's digest. If yes, skip the push for that image. " +
-                  "Default to False. " +
-                  "Note that there is no transactional guarantee between checking for digest existence and pushing the digest. " +
-                  "This means that you should try to avoid running the same container_push targets in parallel.",
-        ),
-        "stamp": STAMP_ATTR,
-        "tag": attr.string(
-            default = "latest",
-            doc = "The tag of the image.",
-        ),
-        "tag_file": attr.label(
-            allow_single_file = True,
-            doc = "The label of the file with tag value. Overrides 'tag'.",
-        ),
-        "tag_tpl": attr.label(
-            mandatory = True,
-            allow_single_file = True,
-            doc = "The script template to use.",
-        ),
-        "windows_paths": attr.bool(
-            mandatory = True,
+        "_pusher": attr.label(
+            default = "//container/go/cmd/pusher",
+            cfg = "host",
```

```suggestion
            cfg = "exec",
```

https://github.com/bazelbuild/bazel/issues/17032